### PR TITLE
Add strict hostname verification information

### PR DESCRIPTION
**Pull request by samgnaniah**
Fixes https://github.com/wso2-enterprise/wso2-apim-internal/issues/726

### CODE REVIEW

File: en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md [hunk 1]
```diff
@@ -255,6 +255,13 @@ been removed from Hotspot JVM.</p>
 <td><p>For password recovery, you can define a regular expression to validate the callback URL. The default configuration allows any callback URL. Note that if you are using the recovery option, it is highly recommended to define the regular expression that validates and only allows access to specific callback URLs.</p>
 </td>
 </tr>
+<tr class="odd">
+<td><p>Strict host verification</p>
+<p><br />
+</p></td>
+<td><p>To configure strict host verification, you can template the `site/public/conf/settings.js` file of each webapp - Publisher, Developer Portal, and Admin. You can template this file using a tool like Ansible if that is what you use now or you can maintain a template file for the settings.js alone as a customization where you can change it through the deployment.toml file.</p>
```
Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md:262]: Let's not mention about Ansible or templating the file as they are going to be customizations only if the customer requires.

Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md:262]: Also shall we reword this to clearly give the path to the settings.js file in each webapp? Something like `site/public/conf/settings.js` in Publisher and Admin portal, `site/public/theme/settings.js` in Developer Portal.

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md [hunk 2]
```diff
@@ -16,6 +16,21 @@ The hostname verification is **disabled** by default. This is done using the `o
     -Dhttpclient.hostnameVerifier=AllowAll \
 ```
 
+When applying the strict hostname verification, you need to change the configuration in the `site/public/conf/settings.js` file of each webapp - Publisher, Developer Portal, and Admin.
+
+In the Developer Portal this path is a little different as it is available in `site/public/theme/setting.js` and the constant is `Settings` instead of `AppConfig` as shown below. 
+
+```javascript
+const AppConfig = {
```
Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/security/enabling-hostname-verification.md:24]: We can remove this const AppConfig = { line altogether to prevent confusion due to 2 different names in webapps. Let's only have the internal body part here.

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md [hunk 3]
```diff
@@ -16,6 +16,21 @@ The hostname verification is **disabled** by default. This is done using the `o
     -Dhttpclient.hostnameVerifier=AllowAll \
 ```
 
+When applying the strict hostname verification, you need to change the configuration in the `site/public/conf/settings.js` file of each webapp - Publisher, Developer Portal, and Admin.
```
Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/security/enabling-hostname-verification.md:19]: Let's give the path clearly here as well.

File: en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md (hunk 1 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md]: Fixed.

Comment (samgnaniah) [reply en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md]: Fixed.

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md (hunk 2 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/security/enabling-hostname-verification.md]: Fixed.

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md (hunk 3 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/security/enabling-hostname-verification.md]: Fixed.

File: en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md [hunk 4]
```diff
@@ -255,6 +255,13 @@ been removed from Hotspot JVM.</p>
 <td><p>For password recovery, you can define a regular expression to validate the callback URL. The default configuration allows any callback URL. Note that if you are using the recovery option, it is highly recommended to define the regular expression that validates and only allows access to specific callback URLs.</p>
 </td>
 </tr>
+<tr class="odd">
```
Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md:258]: There is another section called "Enable HostName Verification" in the same table. I thin we can use it instead of creating a new row. And this already existing row points to [1] which we can make use of here. Then I think no need to add the information in this doc. WDYT?

[1] https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/enabling-hostname-verification/

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md [hunk 5]
```diff
@@ -16,6 +16,18 @@ The hostname verification is **disabled** by default. This is done using the `o
     -Dhttpclient.hostnameVerifier=AllowAll \
 ```
 
+When applying the strict hostname verification, you need to change the configuration in the `site/public/conf/settings.js` file of the Publisher and Admin Portal and the `site/public/theme/setting.js` file of the Developer Portal.
```
Comment (tharikaGitHub) [inline en/docs/install-and-setup/setup/security/enabling-hostname-verification.md:19]: Let's add this as a note and highlight strict hostname verification. Hope that would be fine.

File: en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md (hunk 4 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/deployment-best-practices/security-guidelines-for-production-deployment.md:258]: +1. Will remove this. It's better to only mention in one place anyway.

File: en/docs/install-and-setup/setup/security/enabling-hostname-verification.md (hunk 5 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/security/enabling-hostname-verification.md]: Fixed.

**Review by tharikaGitHub — APPROVED** [review]
